src/Common/src/Interop/Windows/perfcounter/Interop.PerformanceData.cs

```diff
@@ -0,0 +1,93 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal partial class PerfCounter
+    {        
+        [DllImport(Libraries.Advapi32, ExactSpelling = true)]
+        internal static extern unsafe uint PerfStopProvider(
+            IntPtr hProvider
+        );
+
+        internal unsafe delegate uint PERFLIBREQUEST(
+            uint RequestCode,
+            void* Buffer,
+            uint BufferSize
+        );
+
+        // Native PERFLIB V2 Provider APIs.
+        internal struct PerfCounterSetInfoStruct
+        {
+            // PERF_COUNTERSET_INFO structure defined in perflib.h
+            internal Guid CounterSetGuid;
+            internal Guid ProviderGuid;
+            internal uint NumCounters;
+            internal uint InstanceType;
+        }
+
+        internal struct PerfCounterInfoStruct
+        {
+            // PERF_COUNTER_INFO structure defined in perflib.h
+            internal uint CounterId;
+            internal uint CounterType;
+            internal long Attrib;
+            internal uint Size;
+            internal uint DetailLevel;
+            internal uint Scale;
+            internal uint Offset;
+        }
+
+        [StructLayout(LayoutKind.Explicit, Size = 32)]
+        internal struct PerfCounterSetInstanceStruct
+        {
+            // PERF_COUNTERSET_INSTANCE structure defined in perflib.h
+            [FieldOffset(0)] internal Guid CounterSetGuid;
```

Comment:
if i remove this `[FieldOffset(0)]` compiler complain about initialization.

/cc @jkotas @Anipik @JeremyKuhne

Reply:
Where does it complain?

Reply:
`Interop.PerformanceData.cs(52,27): error CS0649: Field 'Interop.PerfCounter.PerfCounterSetInstanceStruct.InstanceNameOffset' is never assigned to, and will always have
its default value 0`

Reply:
on all fields of struct

Reply:
Mark the struct with `[StructLayout(LayoutKind.Sequential)]` attribute. It will make this warning go away.

Reply:
compiler complain only on this struct because is a return val?